test/compiler/inference.jl

```diff
@@ -2998,6 +2998,8 @@ end
 @test ig27907(Int, Int, 1, 0) == 0
 
 # issue #28279
+# ensure that lowering doesn't move these into statement position, which would require renumbering
+using Base: +, -
```

Comment:
This state dependence, where we give very different lowering depending on whether something has been used already in the module, seems very ripe for causing future heisenbugs

Reply:
I agree, but on the other hand, this is supposed to be a storage optimization without semantic effect (though of course there can be bugs).

Reply:
True. I guess in the future we may want to optimize cases like this more aggressively pre-inference too. I just worry about cases where people may pre-lower the code, then ship it off to another process to execute which as slightly different globals at the time. I think Jeff had some code where he was planning to do that and Distributed.jl does that for remotecall.